### PR TITLE
rdma/fi_domain, src/hmem_ze: L0 support

### PR DESCRIPTION
**Pull request by aingerson**
- Add FI_HMEM_ZE to libfabric API
- Add ZE library and header configure check and option for non-default installation location
- Add ZE hmem_ops

### CODE REVIEW

**Comment by aingerson** [comment]
These changes are off the master and will have to be updated if #6077 is merged first.

File: src/hmem_ze.c [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+static inline int _ze_cmd_list_create(ze_device_handle_t device,
+				      ze_command_list_handle_t *cmd_list)
+{
+	ze_command_list_desc_t desc = {
+		.version	= ZE_COMMAND_LIST_DESC_VERSION_CURRENT,
+		.flags		= ZE_COMMAND_LIST_FLAG_NONE,
+	};
+	return zeCommandListCreate(device, &desc, cmd_list);
+}
```
Comment (shefty) [inline src/hmem_ze.c:73]: Both of the above static inline calls wrap a single function and are only called from one place.  They don't really seem useful, and in the queue_create case, it results in initializing desc repeatedly in a loop.

**Comment by aingerson** [comment]
Dependent on #6077 because L0 needs device for copy calls. Will update with create call fixes once that is merged.

**Comment by shefty** [comment]
bot:aws:retest

**Comment by shefty** [comment]
This PR applies and passes CI as is.  Is there a reason to defer merging it?

**Comment by aingerson** [comment]
@shefty I was going to apply your comment about the inlines. I'm surprised this is passing the CI actually since these ZE calls are based on including the device in the copy calls in PR #6077 which ZE needs for the copy.

**Comment by shefty** [comment]
It's probably passing because none of the CI have the Ze libraries installed.

**Comment by aingerson** [comment]
But the code should still be able to compile on the no-ZE path. It would just set the ze.memcpy call to be the ENOSYS defined one
https://github.com/aingerson/libfabric/blob/ze_ops/src/hmem_ze.c#L191

**Comment by aingerson** [comment]
Looks like it's throwing a compile warning. I'm surprised it shows as passing.
```
src/hmem.c:78:19: warning: incompatible pointer types initializing 'int (*)(void *, const void *, size_t)' (aka 'int (*)(void *, const void *, unsigned long)') with an expression of type 'int (void *, const void *, uint64_t, size_t)' (aka 'int (void *, const void *, unsigned long, unsigned long)') [-Wincompatible-pointer-types]
                .copy_to_hmem = ze_hmem_copy,
                                ^~~~~~~~~~~~
src/hmem.c:79:21: warning: incompatible pointer types initializing 'int (*)(void *, const void *, size_t)' (aka 'int (*)(void *, const void *, unsigned long)') with an expression of type 'int (void *, const void *, uint64_t, size_t)' (aka 'int (void *, const void *, unsigned long, unsigned long)') [-Wincompatible-pointer-types]
                .copy_from_hmem = ze_hmem_copy,
                                  ^~~~~~~~~~~~
```

**Comment by shefty** [comment]
bot:aws:retest

**Pull request merged**